app/vmselect/promql/transform.go

```diff
@@ -1988,6 +1991,87 @@ func newTransformFuncSortByLabel(isDesc bool) transformFunc {
 	}
 }
 
+func newTransformFuncAlphaNumericSort(isDesc bool) transformFunc {
+	return func(tfa *transformFuncArg) ([]*timeseries, error) {
+		args := tfa.args
+		if len(args) < 2 {
+			return nil, fmt.Errorf("expecting at least 2 args; got %d args", len(args))
+		}
+		var labels []string
+		for i, arg := range args[1:] {
+			label, err := getString(arg, 1)
+			if err != nil {
+				return nil, fmt.Errorf("cannot parse label #%d for sorting: %w", i+1, err)
+			}
+			labels = append(labels, label)
+		}
+		rvs := args[0]
+		sort.SliceStable(rvs, func(i, j int) bool {
+			for _, label := range labels {
+				a := rvs[i].MetricName.GetTagValue(label)
+				b := rvs[j].MetricName.GetTagValue(label)
+				if string(a) == string(b) {
+					continue
+				}
+				if isDesc {
+					return !alphanumericLess(string(a), string(b))
```

Comment:
nit: may it worth passing slice of runes instead via `bytes.Runes()`? Would that reduce number of allocations?

Reply:
used `bytesutil.ToUnsafeString()`